c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -17,6 +19,617 @@
+    auto complementOfDefault(ecdh);
+    complementOfDefault.insert(complementOfDefault.end(), triple_des.begin(), triple_des.end());
+    aliases.insert({COMPLEMENTOFDEFAULT, complementOfDefault});
+
+    // The content of the default list is determined at compile time and normally corresponds to
+    // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+    aliases.insert({DEFAULT, parse("ALL:!COMPLEMENTOFDEFAULT:!eNULL")});
+  }
+
+  std::vector<Cipher> parse(std::string_view expression) {
+    /**
+     * All ciphers by their openssl alias name.
+     */
+    auto elements = splitCipherSuiteString(std::string(expression));
+    std::vector<Cipher> ciphers;
+    std::vector<Cipher> removedCiphers;
+    for (auto element : elements) {
+
+      if (element.rfind(DELETE, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          remove(&aliases, ciphers, alias);
+        }
+      } else if (element.rfind(EXCLUDE, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          auto toAdd = aliases[alias];
+          removedCiphers.insert(removedCiphers.end(), toAdd.begin(), toAdd.end());
+        }
+      } else if (element.rfind(TO_END, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          moveToEnd(&aliases, ciphers, alias);
+        }
+      } else if ("@STRENGTH" == element) {
+        strengthSort(ciphers);
+        break;
+      } else if (aliases.find(element) != aliases.end()) {
+        add(&aliases, ciphers, element);
+      } else if (element.find(AND) != std::string::npos) {
+        auto intersections = split(element, std::string(1, AND).append("\\"));
+        if (intersections.size() > 0 && aliases.find(intersections[0]) != aliases.end()) {
+          auto result = aliases[intersections[0]];
+          for (int i = 1; i < intersections.size(); i++) {
+            auto alias = aliases.find(intersections[i]);
+            if (alias != aliases.end()) {
+              // make `result` only contain the aliases from `alias`
+              result.erase(std::remove_if(result.begin(), result.end(),
+                                          [&](auto x) {
+                                            return std::find(alias->second.begin(),
+                                                             alias->second.end(),
+                                                             x) != alias->second.end();
+                                          }),
+                           result.end());
+            }
+          }
+          // Add all of `result` onto `ciphers`
+          ciphers.insert(ciphers.end(), result.begin(), result.end());
+        }
+      }
+    }
+    // Remove all ciphers from `ciphers` which are contained in `removedCiphers`
+    ciphers.erase(std::remove_if(ciphers.begin(), ciphers.end(),
+                                 [removedCiphers](Cipher c) {
```

Comment:
I'd reference this vector, instead of copying it into the closure.
```suggestion
                                 [&removedCiphers](Cipher c) {
```